account_invoice_accrual/models/account_invoice.py

```diff
@@ -0,0 +1,297 @@
+                        t[1], self.currency_id)
+                else:
+                    amount_currency = False
+
+                # last line add the diff
+                res_amount_currency -= amount_currency or 0
+                i += 1
+                if i == len(totlines):
+                    amount_currency += res_amount_currency
+
+                iml.append({
+                    'type': 'dest',
+                    'name': name,
+                    'price': t[1],
+                    'account_id': account_id,
+                    'date_maturity': t[0],
+                    'amount_currency': diff_currency_p and
+                    amount_currency or False,
+                    'currency_id': diff_currency_p and
+                    self.currency_id.id or False,
+                    'ref': accrual_ref,
+                })
+        else:
+            iml.append({
+                'type': 'dest',
+                'name': name,
+                'price': total,
+                'account_id': account_id,
+                'date_maturity': self.date_due or False,
+                'amount_currency': diff_currency_p and total_currency or False,
+                'currency_id': diff_currency_p and self.currency_id.id or
+                False,
+                'ref': accrual_ref
+            })
+
+        part = self.env['res.partner']._find_accounting_partner(
+            self.partner_id)
+        line = []
+        for x in iml:
+            vals = self.with_context(accrual_date=accrual_date)\
+                .line_get_convert(x, part.id)
+            line.append((0, 0, vals))
+        line = self.group_lines(iml, line)
+        line = self.finalize_invoice_move_lines(line)
+
+        if move_line_prefix:
+            # update all move name with prefix
+            for ml in line:
+                ml[2]['name'] = ' '.join(
+                    [x for x in [move_line_prefix, ml[2]['name']] if x])
+
+        move = {
+            'ref': accrual_ref,
+            'line_ids': line,
+            'journal_id': accrual_journal_id,
+            'date': accrual_date,
+            'narration': self.comment,
+            'company_id': self.company_id.id,
+            'to_be_reversed': True,
+        }
+
+        accrual_move_id = move_obj.with_context(
+            date=self.date_invoice, invoice=self).\
+            create(move)
+        # make the invoice point to that move
+        self.with_context(
+            date=self.date_invoice, invoice=self).\
+            write({'accrual_move_id': accrual_move_id.id})
+        self._post_accrual_move(accrual_move_id)
+
+        return accrual_move_id.id
+
+    @api.multi
+    def _post_accrual_move(self, accrual_move_id):
+        self.ensure_one()
+        # Pass invoice in context in method post: used if you want to get the
+        # same account move reference when creating the same invoice after a
+        # cancelled one:
+        accrual_move_id.with_context(invoice=self).post()
```

Comment:
Passing the invoice is likely not wanted in this case. Doing this means the accrual move will be take the sequence of your invoice. This means that the invoice will have to get another number. Moreover, if your accrual move is posted in another journal, its sequence will not come from that journal but from the journal of the invoice.
I would replace with
```
# Prevent passing invoice in context in method post: otherwise accrual
# sequence could be the one from this invoice
accrual_move_id.with_context(invoice=False).post()
```